From faa7783e5042ad411a5112fc6c15c5f2f30f1235 Mon Sep 17 00:00:00 2001
From: Vladimir Smirnov <civil@gentoo.org>
Date: Mon, 19 May 2014 19:41:49 +0400
Subject: [PATCH 12/28] Set libperl soname

Bug-Gentoo: https://bugs.gentoo.org/286840

Patch-Name: gentoo/create_libperl_soname.diff

Ported to Perl 5.36.0 dilfridge
---
 Makefile.SH | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/Makefile.SH b/Makefile.SH
index cf11ee4338..8ba0bbf289 100755
--- a/Makefile.SH
+++ b/Makefile.SH
@@ -64,11 +64,11 @@ true)
 				${revision}.${patchlevel}.${subversion}"
 		case "$osvers" in
 	        1[5-9]*|[2-9]*)
-			shrpldflags="$shrpldflags -install_name `pwd`/\$@ -Xlinker -headerpad_max_install_names"
+			shrpldflags="$shrpldflags -install_name `pwd`/libperl.${revision}.${patchlevel}.dylib -Xlinker -headerpad_max_install_names"
 			exeldflags="-Xlinker -headerpad_max_install_names"
 			;;
 		*)
-			shrpldflags="$shrpldflags -install_name \$(shrpdir)/\$@"
+			shrpldflags="$shrpldflags -install_name \$(shrpdir)/libperl.${revision}.${patchlevel}.dylib"
 			;;
 		esac
 		;;
@@ -78,13 +78,15 @@ true)
 		;;
 	sunos*)
 		linklibperl="-lperl"
+		shrpldflags="$shrpldflags -Wl,-soname -Wl,libperl.so.${revision}.${patchlevel}"
 		;;
-	netbsd*|freebsd[234]*|openbsd*|dragonfly*|bitrig*)
+	netbsd*|freebsd*|openbsd*|dragonfly*|bitrig*)
 		linklibperl="-L. -lperl"
+		shrpldflags="$shrpldflags -Wl,-soname -Wl,libperl.so.${revision}.${patchlevel}"
 		;;
 	interix*)
 		linklibperl="-L. -lperl"
-		shrpldflags="$shrpldflags -Wl,--image-base,0x57000000"
+		shrpldflags="$shrpldflags -Wl,--image-base,0x57000000 -Wl,-soname -Wl,libperl.so.${revision}.${patchlevel}"
 		;;
 	aix*)
 		case "$cc" in
@@ -127,6 +129,9 @@ true)
 	           ;;
             esac
             ;;
+	linux*)
+	    shrpldflags="$shrpldflags -Wl,-soname -Wl,libperl.so.${revision}.${patchlevel}"
+	    ;;
 	esac
 	case "$ldlibpthname" in
 	'') ;;
-- 
2.39.3

